lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,428 @@
+
+  /// Does this multireddit require visitors to be over the age of 18.
+  bool get over18 => _data['over_18'];
+
   Multireddit.parse(Reddit reddit, Map data) : super(reddit) {
-    setData(this, data['data']);
+    _data = data['data'];
+    _author = _data['path']?.split('/')[_redditorNameInPathIndex];
+    _infoPath = _generateInfoPath(_data['name'], _author);
+  }
+
+  // Returns valid info_path for multireddit with name `name`.
+  static String _generateInfoPath(String name, String user) =>
+      apiPath['multireddit']
+          .replaceAll(_multiredditRegExp, name)
+          .replaceAll(_userRegExp, user);
+
+  /// Returns a slug version of the [title].
+  static String sluggify(String title) {
+    if (title == null) {
+      return null;
+    }
+    final RegExp _invalidRegExp = new RegExp(r'(\s|\W|_)+');
+    var titleScoped = title.replaceAll(_invalidRegExp, '_').trim();
+    if (titleScoped.length > 21) {
+      titleScoped = titleScoped.substring(21);
+      final lastWord = titleScoped.lastIndexOf('_');
+      //TODO(ckartik): Test this well.
+      if (lastWord > 0) {
+        titleScoped = titleScoped.substring(lastWord);
+      }
+    }
+    return titleScoped ?? '_';
+  }
+
+  /// Add a [Subreddit] to this [Multireddit].
+  ///
+  /// [subreddit] is the name of the [Subreddit] to be added to this [Multireddit].
+  Future add(/* String, Subreddit */ subreddit) async {
+    subreddit = _subredditNameHelper(subreddit);
+    if (subreddit == null) return;
+    final url = apiPath[_kMultiredditUpdate]
+        .replaceAll(_userRegExp, _author)
+        .replaceAll(_multiredditRegExp, displayName)
+        .replaceAll(_subredditRegExp, subreddit);
+    await reddit.put(url, body: {'model': '{"name" : "$subreddit"}'});
+    await refresh();
+  }
+
+  // TODO(@ckartik): Ask @bkonyi if this function should me moved in as a static function
+  // in the [Subreddit] class, and the respective versions of it like [_redditorNameHelper]
+  // in [Subreddit] me moved into the [Redditor] class as a static function.
+  String _subredditNameHelper(/* String, Subreddit */ subreddit) {
+    if (subreddit is Subreddit) {
+      return subreddit.displayName;
+    } else if (subreddit is! String) {
+      throw new DRAWArgumentError('Parameter subreddit must be either a'
+          'String or Subreddit');
+    }
+    return subreddit;
+  }
+
+  /// Copy this [Multireddit], and return the new [Multireddit] of type [Future].
+  ///
+  /// [multiName] is an optional string that will become the display name of the new
+  /// multireddit and be used as the source for the [name]. If [multiName] is not
+  /// provided, the [name] of this [Multireddit] will be used.
+  Future<Multireddit> copy([String multiName]) async {
+    final url = apiPath['multireddit_copy'];
+    final name = sluggify(multiName) ?? _data['display_name'];
+    final userName = await reddit.user.me().then((me) => me.displayName);
+
+    multiName ??= _data['display_name'];
```

Comment:
The analyzer doesn't like reassignment to function parameters so you'll want to create a variable for this.